contracts/WETH10.sol

```diff
@@ -100,27 +79,31 @@ contract WETH10 {
     ///   - caller account must have at least `value` WETH10 token and transfer to account (`to`) cannot cause overflow.
     /// For more information on transferAndCall format, see https://github.com/ethereum/EIPs/issues/677.
     function depositToAndCall(address to, bytes calldata data) external payable returns (bool success) {
-        require(to != address(this), "!recipient");                // This contract is not allowed to hold Wrapped Ether
-        _balanceOf[to] += msg.value;                               // Can never overflow, as this depends on real Ether supply
+        require(to != address(this), "WETH: invalid recipient");
+        balanceOf[to] += msg.value;
+        totalSupply += msg.value;
+        require(totalSupply <= type(uint112).max, "WETH: supply limit exceeded");
         emit Transfer(address(0), to, msg.value);
 
         ERC677Receiver(to).onTokenTransfer(msg.sender, msg.value, data);
         return true;
     }
 
     /// @dev Flash mints WETH10 token and burns from caller account.
+    /// The flash minted WETH10 is not backed by real Ether, but can be withdrawn as such up to the Ether balance of this contract.
     /// Arbitrary data can be passed as a bytes calldata parameter.
     /// Emits two {Transfer} events for minting and burning of the flash minted amount.
     function flashMint(uint256 value, bytes calldata data) external {
-        _balanceOf[msg.sender] += value;
-        require(_balanceOf[msg.sender] >= value, "overflow");
-
+        balanceOf[msg.sender] += value;
+        totalSupply += value;
+        require(totalSupply <= type(uint112).max, "WETH: supply limit exceeded");
```

Comment:
elsewhere i've seen the convention of including the method name in the message which is useful if the message is repeated

`WETH::flashMint: supply limit exceeded`

Reply:
Fixed